### ISSUE TITLE
TST: any_* fixtures are confusingly named

### ISSUE THREAD

**Issue by jreback** [pandas-dev/pandas#38017]
the naming on the any_* fixtures in `pandas/conftest.py` is pretty confusing

- `any_int_dtype` (numpy only)
- `any_int_nullable_dtype` (EA only)
- `any_numeric_dtype` both numpy & EA
- `any_real_dtype` both numpy int & float
- `sint_dtype` signed numpy ints

for example

so we need dtypes that are numpy_only and EA only and numpy + EA both for ints & floats

would propose something like

`any_int_numpy_dtype`
`any_int_ea_dtype`
`any_int_dtype` -> both numpy + ea

same for float

then

`any_numeric_nump_dtype`
`any_numeric_ea_dtype`

for float + int

which means we rename
`sint_dtype` -> `any_signed_int_numpy_dtype`

I think this would require a PR to organize, a PR to create new names & fix everything.

**Comment by jreback** [comment]
cc @jbrockmendel

**Comment by jbrockmendel** [comment]
makes sense

**Comment by jorisvandenbossche** [comment]
Proposed naming seems to make sense

(note I removed the 1.2 milestone, since I don't think this is in any way important for the release? So we can use the milestone to track release critical issues)

**Comment by Mikhaylov-yv** [comment]
take

**Comment by Mikhaylov-yv** [comment]
I apologize, I cannot deal with this problem now.

**Comment by Anupam-USP** [comment]
Hey @jreback I want to take this issue, but being it my very first contribution, how to check the what datatype does the variable takes, like is it numpy or ea or both. Sorry for this silly question

**Comment by jbrockmendel** [comment]
@Anupam-USP look at the `params` in the existing fixture and it should be clear which are numpy dtypes and which are EA dtypes

**Comment by Anupam-USP** [comment]
![](https://user-images.githubusercontent.com/56446640/111285535-3e4e3b80-8667-11eb-9b0c-fc53a61779fd.png)

I think this was the very first dtype mentioned but in the pytest function it is clearly written `ALL_INT_DTYPES`. Is it correct or am I referring to wrong params?

**Comment by Anupam-USP** [comment]
Hey, I am still looking for response on this, not stuck on this but wanted to complete this issue!

**Comment by jbrockmendel** [comment]
Can you rephrase the question?  I don't understand what you're looking for.

**Comment by pratyushsharan** [comment]
@jreback @jbrockmendel Since this issue was most probably lying dormant, I took some liberty and made some changes. Can you review the PR?
